go/client/logclient.go

```diff
@@ -125,20 +125,21 @@ func (c *LogClient) fetchAndParse(uri string, res interface{}) error {
 		return err
 	}
 	resp, err := c.httpClient.Do(req)
-	var body []byte
-	if resp != nil {
-		body, err = ioutil.ReadAll(resp.Body)
-		resp.Body.Close()
-		if err != nil {
-			return err
-		}
-	}
 	if err != nil {
 		return err
 	}
-	if err = json.Unmarshal(body, &res); err != nil {
+	defer resp.Body.Close()
+	// Make sure everything is read, so http.Client can reuse the connection.
+	defer ioutil.ReadAll(resp.Body)
```

Comment:
That's correct. In the case of the HTTP status not being 200, the JSON decoding won't even be attempted, so if we want to reuse the connection, we need to slurp whatever body came in. And when it is attempted, there might be trailing cruft at the end, and if the JSON decoding got everything, it just won't get anything.